src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,171 @@
+            newTask = new Event(description, at);
+
+        } else if (input.startsWith(TODO)) {
+            String description = input.replaceFirst(TODO, "").trim();
+            newTask = new ToDo(description);
+        }
+    }
+
+    /**
+     * Dissects user's raw input and returns an appropriate AddCommand object.
+     * @param input string of user's raw input
+     * @return an appropriate AddCommand object
+     * @throws DukeException
+     */
+    public Command addTask(String input) throws DukeException {
+
+        if (input.startsWith(TODO)) {
+            if (input.substring(4).isEmpty()) {
+                throw new DukeException(TODO_EXCEPTION);
+            }
+            trimInput(input);
+
+        } else if (input.startsWith(DEADLINE)) {
+            if (!input.contains("/by")) {
+                throw new DukeException(DEADLINE_EXCEPTION);
+            } else if (input.substring(input.indexOf("/by") + "/by".length()).isEmpty()) {
+                throw new DukeException(DEADLINE_EMPTY_DESCRIPTION);
+            }
+            trimInput(input);
+
+        } else if (input.startsWith(EVENT)) {
+            if (!input.contains("/at")) {
+                throw new DukeException(EVENT_EXCEPTION);
+            } else if (input.substring(input.indexOf("/at") + "/at".length()).isEmpty()) {
+                throw new DukeException(EVENT_EMPTY_DESCRIPTION);
+            }
+            trimInput(input);
+
+        } else {
+            throw new DukeException(INVALID_TASK_TYPE);
+        }
+
+        return new AddCommand(ui, tasks, newTask);
+    }
+
+    /**
+     * Dissects user's raw input and returns an appropriate SetDoneCommand
+     * @param input string of user's raw input
+     * @return a SetDoneCommand object
+     */
+    public Command markTaskDone(String input) {
+        int taskNumber = Integer.parseInt(input.replace("done ", "")) - 1;
+
+        return new SetDoneCommand(ui, tasks, taskNumber);
+    }
+
+    /**
+     * Dissects user's raw input and returns an appropriate DeleteCommand
+     * @param input string of user's raw input
+     * @return a DeleteCommand object
+     */
+    public Command deleteTask(String input) {
+        int taskNumber = Integer.parseInt(input.replaceFirst("delete", "").trim()) - 1;
+
+        return new DeleteCommand(ui, tasks, taskNumber);
+    }
+
+    /**
+     * Dissects user's raw input beforehand and then calls appropriate command methods.
+     * @param input string of user's raw input
+     * @return a Command object
+     * @throws DukeException
+     */
+    public Command findTask(String input) {
+        String keyword = input.replaceFirst("find", "").trim();
+
+        return new FindCommand(ui, tasks, keyword);
+    }
+
+    public Command selectCommand(String input) throws DukeException {
+
+        String trimmedInput = input.trim().split(" ")[0];
+        Command commandFromInput = null;
+
+        switch (trimmedInput){
+        case TODO: case DEADLINE: case EVENT:
```

Comment:
When you have multiple cases and you want the control to cascade down, you can still follow the same recommended code style to have each case in a new line.